tests/unit/project/test_schema.py

```diff
@@ -281,12 +291,47 @@ def test_invalid_types(self):
 
         raised = self.assertRaises(errors.YamlValidationError, Validator(data).validate)
 
-        expected_message = (
-            "The 'type' property does not match the required "
-            "schema: '{}' is not one of "
-            "['app', 'base', 'gadget', 'kernel', 'snapd']"
-        ).format(self.type_)
-        self.assertThat(raised.message, Equals(expected_message), message=data)
+        self.assertThat(
+            raised.message,
+            # The schema is not loaded in order, so we can get different results depending on what is loaded first.
```

Comment:
This is probably a good example of the use of anyOf schema scenarios leads to less-than-predictable / hard to decipher errors. It really ought to show (all) valid configuration options every time. But understandably that's a non-trivial pivot from here and we should probably put it on the list to revisit longer-term.